bouncer/src/repo/manipulator/modelconvertor/import/repo_model_import_3drepo.cpp

```diff
@@ -285,15 +287,43 @@ RepoModelImport::mesh_data_t RepoModelImport::createMeshRecord(
 
 			uint32_t *tmpIndices = (uint32_t*)(dataBuffer + startEnd[0]);
 
-			for (int i = 0; i < numIndices; i += 3)
+			// figure out primitive type
+			boost::optional<const ptree&> primitiveProp = mesh.get_child_optional(REPO_IMPORT_PRIMITIVE);
+			MeshNode::Primitive primitiveType;
+			if (primitiveProp)
 			{
-				repo_face_t tmpFace;
-
-				tmpFace.push_back(tmpIndices[i]);
-				tmpFace.push_back(tmpIndices[i + 1]);
-				tmpFace.push_back(tmpIndices[i + 2]);
+				primitiveType = static_cast<MeshNode::Primitive>(primitiveProp.get().get_value<int8_t>());
+			}
+			else
+			{
+				primitiveType = MeshNode::Primitive::TRIANGLES;
+			}
 
-				faces.push_back(tmpFace);
+			// pull out faces
+			switch (primitiveType)
+			{
+			case repo::core::model::MeshNode::Primitive::LINES:
```

Comment:
I think it's better to just merge the 2 cases and use variables, as they're essentially doing the same thing hence quite a bit of duplicated code.

ie something like this (untested, unchecked):

```suggestion
			case repo::core::model::MeshNode::Primitive::LINES:
			case repo::core::model::MeshNode::Primitive::TRIANGLES:
			        const auto length = (int8_t)primitiveType;
				for (int i = 0; i < numIndices; i += length)
				{
					repo_face_t tmpFace;
					for (int j = i; j < i + length; ++j)
						tmpFace.push_back(tmpIndices[j]);
					faces.push_back(tmpFace);
				}
				break;
```